### DISCUSSION TITLE
How to manage SD datalogger

### DISCUSSION THREAD

**Discussion by mastraa** [micropython/micropython#14073] [General]
Dear all,

I'm new in programming PICO and Micropython. I'm used to Arduino and its drivers.
However I need to use the SD to log GPS data and I need how to manage uos.mount and uos.umount function.

I have a loop to read and save so I don't really have some point where it save and somewhere not.
How do you suggest to manage it?

I can try to add a 'save'/'not save' button, but I cannot know what happen while I'm debugging code and maybe it will stop between my save/not save toggle.

May you suggest someway to do it?
Is it better to mount/umount everytime i write something in the SD, how does it affect cycle time?

Thank you
Andrea Mastrangelo

**Comment by davefes** [comment]
I can not comment on SD card as I don't use one.

> I have a loop to read and save so I don't really have some point where it save and somewhere not.
How do you suggest to manage it?

I look at the system time 1-2 times a second and on 00 seconds log all my data to a ramdisk (maybe you want data every second)

> I can try to add a 'save'/'not save' button, but I cannot know what happen while I'm debugging code and maybe it will stop between my save/not save toggle.

Then once every hour I save the data to flash.  In your case save it to the SDcard.  You have full control over when things happen.
How often do you want to record GPS data?
